jacodb-approximations/src/main/kotlin/org/jacodb/approximation/ApproximationsMappingFeature.kt

```diff
@@ -0,0 +1,128 @@
+/*
+ *  Copyright 2022 UnitTestBot contributors (utbot.org)
+ * <p>
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ * <p>
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.jacodb.approximation
+
+import org.jacodb.api.ByteCodeIndexer
+import org.jacodb.api.JcClasspath
+import org.jacodb.api.JcDatabase
+import org.jacodb.api.JcFeature
+import org.jacodb.api.JcSignal
+import org.jacodb.api.RegisteredLocation
+import org.jacodb.approximation.annotation.ApproximationFor
+import org.jacodb.impl.fs.className
+import org.jooq.DSLContext
+import org.objectweb.asm.tree.ClassNode
+
+// TODO we need somehow to remove approximation classes from the hierarchy to avoid presence of them in the inheritors of Object
+
+/**
+ *  A feature building mapping between so-called approximations and their targets.
+ *  Other features like [ClassContentApproximationFeature] and [ApproximationsInstructionsFeature]
+ *  uses it for modifying source code of target classes with information from
+ *  the corresponding approximations.
+ *
+ *  Note: for correct work of the feature, you must wait
+ *  till the whole processing of classpath is finished,
+ *  otherwise you might get incomplete mapping.
+ *  See [JcDatabase.awaitBackgroundJobs].
+ */
+object ApproximationsMappingFeature : JcFeature<Any?, Any?> {
+    private val originalToApproximation: MutableMap<OriginalClassName, ApproximationClassName> = mutableMapOf()
```

Comment:
It's better to use CHM since this is global state which is modified from number of threads.